model/alertmanager.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2019-2021 SAP SE or an SAP affiliate company. All rights reserved. This file is
+# licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from model.base import NamedModelElement
+
+
+class AlertManagerConfig(NamedModelElement):
+    def username(self):
+        return self.raw.get('username')
+
+    def password(self):
+        return self.raw.get('password')
+
+    def _required_attributes(self):
+        return (
+            'username',
+            'password',
+        )
```

Comment:
We do already have a base class one could use here.
```suggestion
from model.base import BasicCredentials


class AlertManagerConfig(BasicCredentials):
    def password(self):
        return self.raw.get('password')
```
By default, the provided password-getter would be `passwd()`.

Then again, instead of introducing this boilerplate-class, you _should_ be able to use the BasicCredential class directly in the `config_types.yaml`. I haven't tried that though.

Let's see whether @ccwienk has some thoughts on this :).

Reply:
@neo-liang-sap : as @AndreasBurger indicated, you might configure this type in `config_types.yaml`, like so:

```
alert_manager:
 src:
  file: alertmanager.yaml
 model:
  factory_method: alertmanager
  cfg_type_name: alertmanager
  type: BasicCredentials
```

(in which case you would not need to add the type here). I am however also fine with merging this PR, as is

Reply:
OK i will close this PR and use `BasicCredentials `, thanks!